src/main/java/io/netty/incubator/codec/quic/QuicheError.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.quic;
+
+import io.netty.util.collection.IntObjectHashMap;
+import io.netty.util.collection.IntObjectMap;
+
+/**
+ * All QUIC error codes identified by Quiche. See <a href=
+ * "https://github.com/cloudflare/quiche/blob/0.6.0/src/lib.rs#L335-L380">Error</a>
+ */
+public enum QuicheError {
```

Comment:
Would be cool if we could rename this and make it less implementation dependent. Which would also mean that code should be package-private as the end user should not depend not it.

Maybe rename it `QuicError`.

Reply:
Oh, I thought it's about wrapping Quiche errors specifically. For more general implementation, - sure, - I will rename it. Interesting question here, is the list of errors that we have good enough for general case? E.g. we probably need to remove `DONE` from here, as it's very impl. dependent thing. Maybe something else? 🤔

Reply:
Yes I think Done should be removed as it should never be visible to the end user.

Reply:
The rest should be ok to keep.

That said if our implementation is correct I think `BUFFER_TOO_SHORT` should never happen as well. The rest could be triggered imho


```

Reply:
On it!